lib/ui.coffee

```diff
@@ -15,14 +17,15 @@ module.exports =
     @subs.add @client.onDetached =>
       @ink?.Result.invalidateAll()
 
-    @client.handle progress: (p) =>
-      @progress?.progress = p
+    @client.handle 'progress!': (t, p) => @progress[t] p
 
   deactivate: ->
     @subs.dispose()
+    @progress.clear()
 
   consumeInk: (@ink) ->
     @views.ink = @ink
+    @progress.ink = @ink
 
-    @subs.add @client.onWorking => @progress = @ink.progress.push()
-    @subs.add @client.onDone => @progress?.destroy()
+    @subs.add @client.onWorking => @progress.add progress: -1, id: 'indet'
+    @subs.add @client.onDone => @progress.clear()
```

Comment:
Hm, not sure how to handle this then. We could of course only destroy the `indet` progress bar `onDone`, but that'll leave progress bars on the stack when e.g. interrupting a loop (which leads to progress bars not being destroyed).